src/ParaTest/Coverage/CoverageMerger.php

```diff
@@ -2,43 +2,80 @@
 
 namespace ParaTest\Coverage;
 
+use SebastianBergmann\CodeCoverage\CodeCoverage;
+
 class CoverageMerger
 {
     /**
-     * @var \PHP_CodeCoverage
+     * @var \PHP_CodeCoverage|CodeCoverage
      */
     private $coverage = null;
 
-    public function addCoverage(\PHP_CodeCoverage $coverage)
+    /**
+     * @param \PHP_CodeCoverage|CodeCoverage $coverage
+     */
+    private function addCoverage($coverage)
     {
-        if ($this->coverage == null) {
-            $this->setCoverage($coverage);
+        if (null === $this->coverage) {
+            $this->coverage = $coverage;
         } else {
-            $this->mergeCoverage($coverage);
+            $this->coverage->merge($coverage);
         }
     }
 
     /**
-     * @param \PHP_CodeCoverage $coverage
+     * Returns coverage object from file.
+     *
+     * @param \SplFileObject $coverageFile Coverage file.
+     *
+     * @return \PHP_CodeCoverage|CodeCoverage
      */
-    private function setCoverage(\PHP_CodeCoverage $coverage)
+    private function getCoverageObject(\SplFileObject $coverageFile)
     {
-        $this->coverage = unserialize(serialize($coverage));
```

Comment:
I've removed this `unserialize(serialize())` trick. Did not find any comment why it is used and since tests are passing with different php and phpunit versions decided to remove it cause it can reduce performance.